packages/trader/src/Modules/Account/Sections/Security/LoginHistory/login-history.jsx

```diff
@@ -0,0 +1,169 @@
+    fields = {
+        date: localize('Date and time'),
+        action: localize('Action'),
+        browser: localize('Browser'),
+        ip: localize('IP address'),
+        status: localize('Status'),
+    };
+    // state
+    state = {
+        is_loading: true,
+        fetch_limit: 12, // TODO: put it in constants or configs
+        error: '',
+        data: [],
+    };
+    // methods
+    getLoginHistoryColumnsTemplate = () =>
+        Object.keys(this.fields).map(key => ({ title: this.fields[key], col_index: key }));
+
+    getFormattedData(login_history) {
+        const data = [];
+        for (let i = 0; i < this.state.fetch_limit; i++) {
+            data[i] = {};
+            const environment = login_history[i].environment;
+            const environment_split = environment.split(' ');
+            const date = environment_split[0];
+            const time = environment_split[1].replace('GMT', '');
+            const date_time = convertDateFormat(`${date} ${time}`, 'D-MMMM-YY hh:mm:ss', 'YYYY-MM-DD hh:mm:ss');
+            data[i].date = `${date_time} GMT`;
+            data[i].action = login_history[i].action;
+            const user_agent = environment.substring(environment.indexOf('User_AGENT'), environment.indexOf('LANG'));
+            const ua = parseUA(user_agent);
+            data[i].browser = ua ? `${ua.name} ${ua.version}` : '';
+            data[i].ip = environment_split[2].split('=')[1];
+            data[i].status = login_history[i].status === 1 ? localize('Success') : localize('Failure');
+            data[i].id = i;
+        }
+        return data;
+    }
+
+    static getLoginHistory(limit) {
+        return new Promise(resolve => {
+            WS.authorized.loginHistory(limit).then(data => {
+                if (data.error) resolve({ api_initial_load_error: data.error.message });
+                resolve(data);
+            });
+        });
+    }
+    // lifecycle methods
+    async componentDidMount() {
+        const api_res = await LoginHistory.getLoginHistory(this.state.fetch_limit);
+        if (api_res.api_initial_load_error) {
+            this.setState({
+                is_loading: false,
+                error: api_res.api_initial_load_error,
+            });
+        } else {
+            const formated_data = this.getFormattedData(api_res.login_history);
+            this.setState({
+                is_loading: false,
+                data: formated_data,
+            });
+        }
+    }
+
+    render() {
+        if (this.props.is_switching) return <Loading />;
+        if (this.state.is_loading) return <Loading is_fullscreen={false} className='account___intial-loader' />;
+
+        if (this.state.error) return <LoadErrorMessage error_message={this.state.error} />;
+
+        this.columns = this.getLoginHistoryColumnsTemplate();
+
+        return (
+            <section className='login-history-container'>
+                {this.state.data && this.state.data.length > 0 ? (
```

Comment:
no need for ternary, could just be `&&`

Reply:
Plus the `?` operator
```suggestion
                {this.state.data?.length > 0 && (
```